dbcon/joblist/tupleaggregatestep.cpp

```diff
@@ -5192,6 +5194,25 @@ void TupleAggregateStep::aggregateRowGroups()
     }
 }
 
+void TupleAggregateStep::threadedAggregateFinalize(uint32_t threadID)
+{
+  for (uint32_t i = 0; i < fNumOfBuckets; ++i)
+  {
+    if (fAgg_mutex[i]->try_lock())
```

Comment:
I am curios about the reasons why try_lock() and not lock() is used here?

Reply:
We go through the bucket list and take the first one that isn't currently being finalized, so there is no need to wait for the processing to complete. The state of the race here is resolved by the flag in RowAggStorage.